pep-0688.rst

```diff
@@ -209,41 +279,45 @@ more consistent, so we believe the migration cost is worth it.
 How to Teach This
 =================
 
-We will add notes pointing to ``types.Buffer`` in appropriate places in the
+We will add notes pointing to ``collections.abc.Buffer`` in appropriate places in the
 documentation, such as `typing.readthedocs.io <https://typing.readthedocs.io/en/latest/>`__
 and the `mypy cheat sheet <https://mypy.readthedocs.io/en/stable/cheat_sheet_py3.html>`__.
 Type checkers may provide additional pointers in their error messages. For example,
 when they encounter a buffer object being passed to a function that
 is annotated to only accept ``bytes``, the error message could include a note suggesting
-the use of ``types.Buffer`` instead.
+the use of ``collections.abc.Buffer`` instead.
 
 
 Reference Implementation
 ========================
 
-An implementation of ``types.Buffer`` is
-`available <https://github.com/python/cpython/compare/main...JelleZijlstra:typesbuffer?expand=1>`__
+An implementation of this PEP is
+`available <https://github.com/python/cpython/compare/main...JelleZijlstra:pep688v2?expand=1>`__
 in the author's fork.
 
 
 Rejected Ideas
 ==============
 
-Buffer ABC
-----------
+types.Buffer
+------------
 
-An `earlier proposal <https://github.com/python/cpython/issues/71688>`__ suggested
-adding a ``collections.abc.Buffer``
-`abstract base class <https://docs.python.org/3/glossary.html#term-abstract-base-class>`__
-to represent buffer objects. This idea
-stalled because an ABC with no methods does not fit well into the ``collections.abc``
-module. Furthermore, it required manual registration of buffer classes, including
-those in the standard library. This PEP's approach of using the ``__instancecheck__``
-hook is more natural and does not require explicit registration.
+An earlier version of this PEP proposed adding a new ``types.Buffer`` type with
+an ``__instancecheck__`` implemented in C so that ``isinstance()`` checks can be
+used to check whether a type implements the buffer protocol. This avoids the
+complexity of exposing the full buffer protocol to Python code, while still
+allowing the type system to check for the buffer protocol.
 
-Nevertheless, the ABC proposal has the advantage that it does not require C changes.
-This PEP proposes to adopt a version of it in the third-party ``typing_extensions``
-package for the benefit of users of older Python versions.
+However, this approach
+does not compose well with the rest of the type system, because ``types.Buffer``
+would be a nominal type, not a structural one. For example, there would be no way
+to represent "an object that supports both the buffer protocol and ``__len__``". With
+the current proposal, the ``__buffer__`` class is like any other method, so a
```

Comment:
This is the first time you've referred to the class as "the ``__buffer__`` class" in this document :)